### ISSUE TITLE
NAK causes n_dhcp4_client_lease_select failure

### ISSUE THREAD

**Issue by gotthardp** [nettools/n-dhcp4#30]
When the client is in STATE_RENEWING, there is a (current) `probe->current_lease`.
Now, when a NAK arrives the state changes (in `n_dhcp4_client_probe_transition_nak`) to STATE_INIT, retaining the current lease.
Then, once the DISCOVER-OFFER-REQUEST-ACK dance is over, the `n_dhcp4_client_lease_select` gets called. It checks the `probe->current_lease` is (still) not NULL and fails. The probe is not able to recover from this.

I wonder-- shouldn't be the `probe->current_lease` cleared in `n_dhcp4_client_probe_transition_nak` just like it is cleared in `n_dhcp4_client_probe_transition_lifetime`? Or should the custom code unref it prior calling the `n_dhcp4_client_lease_select`? (I will create a PR if you confirm this is a bug to be fixed here.)

**Comment by bengal** [comment]
Good catch, this was fixed in e4af93228e3772bbb443ec1237252e6a2f3e3dd7.
